tremor-script/src/interpreter.rs

```diff
@@ -693,99 +741,138 @@ where
     'event: 'run,
 {
     let patch_expr = expr;
+    let mut evaluated: Vec<PreEvaluatedPatchOperation> = Vec::with_capacity(expr.operations.len());
```

Comment:
Is there a specific reason you chose to go for
```rust
let mut coll : Vec<_> = Vec::with_capacity(a.len());
for op in &a {
   coll.push( /* transform op */ );
}
```
instead of
```rust
let coll : Vec<_> = a.iter()
   .map(|op| /* transform op */)
   .collect();
```
?

Reply:
the reason is the error handling we would need to put into map. So it is a convenience thing, otherwise we would have to map every element to a result and error on the first one we find. With a for loop we can error on first occurrence.